test/e2e/new-dns-record.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('not authenticated', () => {
+  test('redirects to login page', async ({ page }) => {
+    await page.goto('/');
+    await expect(page).toHaveURL(/.*login.*/);
+  });
+});
+
+test.describe('authenticated as user', () => {
+  // Use shared state for being logged in as user
+  test.use({ storageState: 'test/e2e/.auth/user.json' });
+
+  test('redirects to edit dns record page when required fields are filled', async ({ page }) => {
+    await page.goto('/domains/new');
+    await page.getByLabel('Record Name*').fill('test');
+    await page.getByRole('combobox', { name: 'Type' }).selectOption('A');
+    await page.getByLabel('Value*').fill('test');
+    await page.getByRole('button', { name: 'Create' }).click();
+    await expect(page).toHaveURL(/.*domains\/[0-9]*/);
+  });
+
+  test('redirects to edit dns record page when all fields are filled', async ({ page }) => {
+    await page.goto('/domains/new');
+    await page.getByLabel('Record Name*').fill('test');
+    await page.getByRole('combobox', { name: 'Type' }).selectOption('A');
+    await page.getByLabel('Value*').fill('test');
+    await page.getByLabel('Ports').fill('port1, port2');
+    await page.getByLabel('Course').fill('test course');
+    await page.getByLabel('Description').fill('test description');
+    await page.getByRole('button', { name: 'Create' }).click();
+    await expect(page).toHaveURL(/.*domains\/[0-9]*/);
+  });
+
+  test('does not create dns record if required fields are empty', async ({ page }) => {
+    await page.goto('/domains/new');
+    await page.getByRole('button', { name: 'Create' }).click();
+    await expect(page).toHaveURL(/.*domains\/new/);
```

Comment:
Do we want to check for an error message in the page?

Reply:
There isn't any way to check the error messages since front end validation exists through Chakra UI components:
<img width="505" alt="image" src="https://user-images.githubusercontent.com/67077705/224379669-e146acf6-8555-4d12-be36-d38535b34b6b.png">
I tried to grab it but I am pretty sure it is not possible

Reply:
In a follow-up, we could [style the control](https://chakra-ui.com/docs/components/input#changing-the-focus-and-error-border-colors) and check for that I guess.